### PR TITLE
Improve the documentation on how to get started with the Java SDK

### LINKED ISSUE #1595: Improve the documentation on how to get started with the Java SDK
### Problem

Multiple community members have submitted issues where their projects are failing due to incorrect setup. For reference, see:
- **Java**: #1614 and #1615
- **Android**: https://github.com/hashgraph/hedera-sdk-java/issues/1566 and [this commit](https://github.com/aadityapaliwal/WalletConnectDemo/compare/main...1566).

### Solution

To improve the documentation on how to get started with the Java SDK

### PR DESCRIPTION
**Pull request by thenswan**
**Description**:
Improve the documentation on how to get started with the Java SDK:
- improves documentation structure;
- simplifies the documentation;
- adds documentation based on user issues;
 
This PR also fixes the documentation:
- min Android SDK level;
- running SDK integration tests;
- etc.

**Related issue(s)**:
Fixes #1595 #1622

### CODE REVIEW

**Review by thenswan — COMMENTED** [review]
My questions for review

File: docs/android-app/android-app-quickstart.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+## Get started
```
[the remark this answers is not in the document]
Comment (rokn) [reply docs/android-app/android-app-quickstart.md:1]: Yeah I don't think we need this

File: example-android/README.md [hunk 2]
```diff
@@ -1,17 +1,50 @@
...
 ```xml
 <string name="operator_id">...</string>
 <string name="operator_key">...</string>
 ```
 
-## Running with a local SDK version
-
-By default, the example uses the SDK version published on Maven Central. If you want to use the local version you need to do this:
-
+### Running with a local SDK version
```
[the remark this answers is not in the document]
Comment (rokn) [reply example-android/README.md:30]: That's more of a dev thing, I think here it's fine

File: docs/sdk/developer-guide.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+$ ./gradlew compileJava
+```
+
+### Unit Tests
+
+```sh
+$ ./gradlew test
+```
+
+### Integration Tests
```
[the remark this answers is not in the document]
Comment (rokn) [reply docs/sdk/developer-guide.md:18]: If it's not straightforward let's do it

File: docs/java-app/java-app-quickstart.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
+## Get started
```
[the remark this answers is not in the document]
Comment (thenswan) [reply docs/java-app/java-app-quickstart.md:1]: Keep as it is now.

File: docs/sdk/developer-guide.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (thenswan) [reply docs/sdk/developer-guide.md:18]: @rokn it needs its own issue

[the remark this answers is not in the document]
Comment (thenswan) [reply docs/sdk/developer-guide.md:18]: Done

**Review by rokn — APPROVED** [review]

**Review by Petyo-Lukanov — APPROVED** [review]

**Review by agadzhalov — APPROVED** [review]

**Review by agadzhalov — APPROVED** [review]
LGTM

**Pull request merged**